### PR TITLE
search added to activities page, fixes #464 : addition to #344

### LINKED ISSUE #464: Search feature missing from activities.html
### Expected results
- feature to search issues 

### Actual results
- all open issues seen

### Steps to reproduce the issue
-add a search feature

## Would you like to work on this
- woring 🔥

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Thank you for your pull request! Please read and understand everything below

Please acknowledge that you have read and understood each of the following before submitting your pull request.

# _Do not_ delete any text other than where you are instructed.

**Maintainers: If any of these are left unchecked, do not merge.**

**Students: If one of them is not applicable to you -- Please check it anyways. DO NOT REMOVE ANYTHING!**

**You may check them after opening your pull request.**

You may do so by changing each `[ ]` to `[x]` Please take note of the whitespace as it matters.

- [x] Read and understood (see CONTRIBUTING.md)
- [x] Included a live link or preview screenshot
- [ ] Images are `240 x 240` [w x h]
- [x] Resolved merge conflicts
- [x] Included a description of my change below

# Things done in this Pull Request
- added lazy load for faster loading of 3 images

https://yashkumarverma.github.io/gci17.fossasia.org/
🔥

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
Can you please share demo link

**Comment by YashKumarVerma** [comment]
@vaibhavsingh97 https://yashkumarverma.github.io/gci17.fossasia.org/

File: activities.html [hunk 1]
```diff
@@ -10,10 +10,17 @@
...
+			</div>
+		</div>
 		<div class="col-sm-12 col-md-12 text-center mentor-list" id="issues">
 			<div class="issue-container container"></div>
 		</div>
 	</div>
 </section>
 
 {% include footer.html %}
+
```
Comment (abishekvashok) [inline activities.html]: You got spaces here

Comment (YashKumarVerma) [reply activities.html]: something big coming up. ignore the PR for a few minutes 🔥

**Comment by YashKumarVerma** [comment]
https://yashkumarverma.github.io/gci17.fossasia.org/activities

**Comment by YashKumarVerma** [comment]
@abishekvashok @vaibhavsingh97

**Review by 0x48piraj — CHANGES_REQUESTED** [review]
image

Your implemented Search bar is very tenuous :-1: 
Please improve that.

---

Your patch works @YashKumarVerma 
Tested it.  Improves browser caching from Grade B to A :+1:

File: activities.html [hunk 2]
```diff
@@ -10,6 +10,12 @@
 			<h2 class="animated fadeIn">Website Activity</h2>
 			<p class="lead mb64">Things to work on with</p>
 		</div>
+		<div class="col-sm-12">
+			<div class="form-group">
+			<input type="text" id="searchString" class="form-control" placeholder="Search for an issue ..." />
+			<div id="issueSearchOutput"></div>
```
Comment (vaibhavsingh97) [inline activities.html]: Do we need seperate division for declaration of ID?

Comment (vaibhavsingh97) [reply activities.html]: `id` can be declared in input itself

File: js/scripts.js [hunk 3]
```diff
@@ -1456,3 +1456,23 @@ window.onload = function() {
...
+    for (var j=0; j<strArray.length; j++) {
+        if (strArray[j][1].title.match(str)) return j;
+    }
+    return -1;
+}
+
+$("#searchString").keydown(function(){
+    // issueArr has all issues loaded in contributers.js
+    var string = $("#searchString").val();
+    var ele = searchStringInArray(string,issueArr);
```
Comment (vaibhavsingh97) [inline js/scripts.js]: Use descriptive variable name instead of `ele`

File: js/scripts.js [hunk 4]
```diff
@@ -1456,3 +1456,23 @@ window.onload = function() {
...
+    if(ele != -1){
+        var output = "Searching for <b><a target='_BLANK' href='"+issueArr[ele][1].html_url+"'>"+"#"+issueArr[ele][1].number+" "+issueArr[ele][1].title+" by "+issueArr[ele][1].user.login+"</a></b>";
```
Comment (vaibhavsingh97) [inline js/scripts.js]: replace `_BLANK` w/  `_blank`

**Review by vaibhavsingh97 — CHANGES_REQUESTED** [review]
Good Job!! keep it up 🎖 
As an enhacement do what @0x48piraj had asked and also instead of adding 
image
you can just load one of these 
image
elements 👍 
You can always update this PR

**Comment by nzec** [comment]
@YashKumarVerma Maybe use a autocomplete/search plugin.

**Comment by YashKumarVerma** [comment]
@duttaditya18 @0x48piraj @vaibhavsingh97 @abishekvashok 
Updated 🔥

**Review by nzec — APPROVED** [review]
@YashKumarVerma Looks good! 😎 

Only problem I see is that search does not work if the case is not exactly the same. I would suggest you to convert all the project names and the searched terms to lowercase and then try to find it. 

But all that can be added later. (Sorry if I have been babbling too much. 😄 )

And Merge conflicts as well! 👍

**Comment by YashKumarVerma** [comment]
@duttaditya18  No issues at all. It makes our site better ❇️ 
image
Fixed.

**Comment by anshumanv** [comment]
Amazing work really! Can we somehow display all matching results? Presently it shows just one AFAIK. :smile:

**Comment by YashKumarVerma** [comment]
@anshumanv we have a complete list of all the issues down there. is it really required 😕

**Comment by YashKumarVerma** [comment]
can we have it merged and feature improvements be done later on ?

**Comment by nzec** [comment]
@anshumanv I also think this should be merged and then new issues can be opened for improvements to this,

**Comment by YashKumarVerma** [comment]
@vaibhavsingh97 can we merge this now ? 

i have to open up a new pr for a new issue too 🔥

**Comment by vaibhavsingh97** [comment]
@YashKumarVerma I mean to say display only those issue which matches the user query. I am merging this PR and will send new PR as an improvement. (Don't forget to open new issue 😉 )

**Review by vaibhavsingh97 — APPROVED** [review]

**Pull request merged**